### PR TITLE
Error when running generate with --steps

### LINKED ISSUE #109: Error when running generate with --steps
```
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /usr/local/lib/python3.10/site-packages/kpops/cli/main.py:224 in generate    │
│                                                                              │
│   221 │   │   for component in steps_to_apply:                               │
│   222 │   │   │   component.template(api_version, ca_file, cert_file)        │
│   223 │   elif cert_file or ca_file or api_version or steps:                 │
│ ❱ 224 │   │   raise TypeError(                                               │
│   225 │   │   │   "The following flags can only be used in conjuction with ` │
│   226 │   │   │   │   '--cert-file'\n \                                      │
│   227 │   │   │   │   '--ca-file'\n \                                        │
╰──────────────────────────────────────────────────────────────────────────────╯
TypeError: The following flags can only be used in conjuction with `--template`:
                 '--cert-file'
                 '--ca-file'
                 '--api-version'
                 '--steps'
```

### PR DESCRIPTION
**Pull request by sujuka99**
fixes #109

### CODE REVIEW

**Comment by sujuka99** [comment]
Can the warning also cause issues? If yes, I will just remove the else entirely.

File: kpops/cli/main.py [hunk 1]
```diff
@@ -220,8 +220,8 @@ def generate(
         for component in steps_to_apply:
             component.template(api_version, ca_file, cert_file)
     elif cert_file or ca_file or api_version or steps:
-        raise TypeError(
-            "The following flags can only be used in conjuction with `--template`: \n \
+        log.warning(
```
Comment (sujuka99) [inline kpops/cli/main.py:223]: Can the warning also cause issues? If yes, I will just remove the else entirely.

**Review by philipp94831 — COMMENTED** [review]
Wouldn't it be better to also just filter the steps when generating the output?

**Comment by sujuka99** [comment]
I think it's better to release a patch as soon as possible as KPOps is supposed to be stable now.

**Comment by philipp94831** [comment]
I don't think anything is really broken. Just the behavior was different some time ago. But if people are just starting to use it now, all is fine

File: kpops/cli/main.py (hunk 1 above)
Comment (raminqaf) [reply kpops/cli/main.py:223]: No the warning does not cause an issue :)

**Comment by raminqaf** [comment]
> I don't think a quick fix is needed. Just #110 :-)

@sujuka99 I agree with @philipp94831. The fix for this bug is implementing #110. As far as I know. None of the projects are facing this bug now. You can go ahead and merge this because I still think that throwing an exception is unnecessary.

**Review by raminqaf — APPROVED** [review]

**Pull request merged**